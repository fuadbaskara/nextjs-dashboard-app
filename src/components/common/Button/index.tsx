import React from 'react';

export type ButtonProps = {
  children: React.ReactNode;
  [key: string]: any;
};

function Button({ children, ...props }: ButtonProps) {
  return (
    <button type="button" {...props}>
      {children}
    </button>
  );
}

export default Button;
