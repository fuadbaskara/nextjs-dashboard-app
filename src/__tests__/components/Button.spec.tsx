import React from 'react';
import { screen } from '@testing-library/react';
import Button from '@/components/common/Button';
import { render } from '../test-utils';

const renderBtn = () => render(<Button>Click me</Button>);

describe('Button Component', () => {
  test('renders correctly with simple props:', () => {
    const { asFragment } = renderBtn();
    expect(screen.getByText('Click me')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
