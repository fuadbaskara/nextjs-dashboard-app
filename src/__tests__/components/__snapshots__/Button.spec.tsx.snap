// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly with simple props: 1`] = `
<DocumentFragment>
  <button
    type="button"
  >
    Click me
  </button>
</DocumentFragment>
`;
